app/routes/session.js

```diff
@@ -61,6 +61,20 @@ function SessionHandler(db) {
             var invalidPasswordErrorMessage = "Invalid password";
             if (err) {
                 if (err.noSuchUser) {
+                    console.log('Error: attempt to login with invalid user: ', userName);
+
+                    // Fix for A1 - 3 Log Injection - encode/sanitize input for CSRF Injection
```

Comment:
Would this be  a CSRF or CRLF Injection?

Reply:
You're totally right on the CSRF thing - I meant to write [CRLF](https://www.owasp.org/index.php/CRLF_Injection) but my brain autocompleted the wrong thing :-)